test/routs/soldiersTest.js

```diff
@@ -0,0 +1,174 @@
+const mocha = require("mocha");
+const http = require("http");
+const XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+const expect = require('chai').expect;
+const configs = require('../../config');
+const fs = require("fs");
+const runServer = require("../../start").runServer;
+
+let mongoServer;
+let client;
+let db;
+
+describe("soldiers api", function () {
+  const url = configs.serverUrl + 'soldiers';
+  let testData = 1;
+
+  before("soldiers api", function (done) {
+    runServer(function (server, clientOut, dbOut) {
+      mongoServer = server;
+      client = clientOut;
```

Comment:
to maake it clearer that you are saving them on the context. use this.client = ....